lib/services/local/trust.go

```diff
@@ -160,48 +218,74 @@ func (s *CA) CompareAndSwapCertAuthority(new, expected types.CertAuthority) erro
 
 // DeleteCertAuthority deletes particular certificate authority
 func (s *CA) DeleteCertAuthority(ctx context.Context, id types.CertAuthID) error {
-	if err := id.Check(); err != nil {
-		return trace.Wrap(err)
-	}
-	// when removing a types.CertAuthority also remove any deactivated
-	// types.CertAuthority as well if they exist.
-	err := s.Delete(ctx, backend.Key(authoritiesPrefix, deactivatedPrefix, string(id.Type), id.DomainName))
-	if err != nil {
-		if !trace.IsNotFound(err) {
+	return s.DeleteCertAuthorities(ctx, id)
```

Comment:
We haven't optimized dynamodbbk to do a regular write if there's only one action, have we? 😬

Reply:
No, but IMO it's not a big deal.  Way I see it, any operation that we anticipate being called frequently enough for such an optimization to be impactful is also too hot of an operation for us to be using `AtomicWrite` at all.

Reply:
I also just realized that even with a single type we're still doing an atomic delete on both the active and inactive keys for the CA, so my original point is moot anyway.

Are we ok with breaking the convention of DeleteFoo returning NotFound if the Foo doesn't exist? I can't really think of a reason why we'd ever care about a NotFound on a "single" CA, but maybe there's fringe cases where that's relevant?

Reply:
I tried to find a case where we care, and couldn't.  I did actually write a janky version of this that first reads the various CA locations and sets up assertions accordingly, but it seemed like a lot of logic for something that none of our actual usage wants (all our existing CA logic was explicitly ignoring `NotFound` errors).  Also, it was fallible to concurrent "moves".  Feels like doing conditional deletes of values that may be concurrently moved from one key to another needs an `AtomicRead` in order to really be ergonomic.